app/Http/Controllers/V2/AuditStatus/StoreAuditStatusController.php

```diff
@@ -4,82 +4,47 @@
-use Illuminate\Support\Facades\Storage;
 
 class StoreAuditStatusController extends Controller
 {
     use SaveAuditStatusTrait;
 
-    public function __invoke(Request $request): AuditStatusResource
+    public function __invoke(AuditStatusCreateRequest $auditStatusCreateRequest): AuditStatusResource
     {
-        $body = $request->all();
+        $body = $auditStatusCreateRequest->all();
 
-        $auditable_type = $this->getAuditableTypeFromRequest($request);
+        $model = $this->getEntityFromRequest($auditStatusCreateRequest);
         if ($body['type'] === 'change-request') {
             AuditStatus::where([
-                ['auditable_id', $body['auditable_id']],
+                ['auditable_id', $model->id],
                 ['type', 'change-request'],
                 ['is_active', true],
             ])->update(['is_active' => false]);
-            $auditStatusresponse = $this->saveAuditStatus($auditable_type, $body['auditable_id'], $body['status'], $body['comment'], $body['type'], $body['is_active'], $body['request_removed']);
+            $auditStatus = $this->saveAuditStatus(get_class($model), $model->id, $body['status'], $body['comment'], $body['type'], $body['is_active'], $body['request_removed']);
         } else {
-            $auditStatusresponse = $this->saveAuditStatus($auditable_type, $body['auditable_id'], $body['status'], $body['comment'], $body['type']);
+            $auditStatus = $this->saveAuditStatus(get_class($model), $model->id, $body['status'], $body['comment'], $body['type']);
         }
+        $auditStatus->entity_name = $model->name;
 
-        if ($request->file('file')) {
-
-            foreach ($request->file('file') as $file) {
-                if ($file !== null) {
-                    try {
-                        $this->saveAuditAttachments($file, $auditStatusresponse);
-                    } catch (\Exception $e) {
-                        Log::error('Error uploading file: ' . $e->getMessage());
-
-                        throw $e;
-                    }
-                }
-            }
-        }
-
-        return new AuditStatusResource($auditStatusresponse);
-    }
-
-    private function saveAuditAttachments($file, $auditStatusresponse)
-    {
-        $fileName = time() . $file->getClientOriginalName();
-        $filePath = 'public/' . $fileName;
-
-        Storage::disk('s3')->put($filePath, file_get_contents($file), 'public');
-        $urlPath = Storage::disk('s3')->url($filePath);
-
-        AuditAttachment::create([
-            'auditable_id' => $auditStatusresponse->id,
-            'file_name' => $file->getClientOriginalName(),
-            'file_url' => $urlPath,
-            'date_created' => now(),
-            'created_by' => Auth::user()->email_address,
-        ]);
+        return new AuditStatusResource($auditStatus);
     }
 
-    private function getAuditableTypeFromRequest(Request $request)
+    private function getEntityFromRequest(AuditStatusCreateRequest $request)
```

Comment:
This method should not be needed if the `ModelBindingInterfaceMiddleware` is used for this endpoint as well.